src/js/actions/deviceActions.js

```diff
@@ -908,3 +933,47 @@ export const setDeviceTags = (deviceId, tags) => dispatch =>
       .catch(err => commonErrorHandler(err, `There was an error setting tags for device ${deviceId}.`, dispatch, 'Please check your connection.'))
       .then(() => Promise.resolve(dispatch({ type: DeviceConstants.RECEIVE_DEVICE, device: { ...device, tags } })));
   });
+
+export const getDeviceTwin = (deviceId, integrationProvider) => (dispatch, getState) =>
+  GeneralApi.get(`${apiBase}${DeviceConstants.EXTERNAL_PROVIDER[integrationProvider].managementUrl}/device/${deviceId}/twin`)
+    .catch(err => commonErrorHandler(err, `There was an error getting the device twin for device ${deviceId}.`, dispatch))
+    .then(({ data }) =>
+      Promise.resolve(
+        dispatch({
+          type: DeviceConstants.RECEIVE_DEVICE,
+          device: {
+            ...getState().devices.byId[deviceId],
+            twinsByProvider: {
+              ...getState().devices.byId[deviceId].twinsByProvider,
+              [integrationProvider]: {
+                ...data.properties,
+                updated_ts: data.lastActivityTime
+              }
+            }
+          }
+        })
+      )
+    );
+
+export const setDeviceTwin = (deviceId, integrationProvider, settings) => (dispatch, getState) =>
+  GeneralApi.put(`${apiBase}${DeviceConstants.EXTERNAL_PROVIDER[integrationProvider].managementUrl}/device/${deviceId}/twin`, settings)
```

Comment:
```suggestion
  GeneralApi.put(`${apiBase}${DeviceConstants.EXTERNAL_PROVIDER[integrationProvider].managementUrl}/devices/${deviceId}/twin`, settings)
```

Reply:
should now point to the right endpoint 👌